controllers/src/main/java/com/redhat/ipaas/controllers/integration/IntegrationController.java

```diff
@@ -74,13 +72,17 @@ public void start() {
             if (event!=null && "change-event".equals(event)) {
                 try {
                     ChangeEvent changeEvent = Json.mapper().readValue(data, ChangeEvent.class);
-                    if ((changeEvent != null
-                        || changeEvent.getKind().isPresent() && changeEvent.getId().isPresent())
-                        && (Kind.from(changeEvent.getKind().get()) == Kind.Integration)) {
-                        enqueueCheckIntegration(changeEvent.getId().get());
+                    if (changeEvent != null) {
+                        changeEvent.getId().ifPresent(
+                            id -> changeEvent.getKind().ifPresent(
+                                kind -> {
+                                    if (Kind.from(kind) == Kind.Integration) {
+                                        enqueueCheckIntegration(id);
```

Comment:
I think, too, that it makes sense to have a central and global loop and don't let the handlers themselves manage the scheduling. This would avoid situations likes this.

As this would be some major change (post summit ?), I suggest as a quick fix here to only react on changes in non monitored integrations. @chirino added already something for this, but it seems not to work yet properly.